x/merkledb/history.go

```diff
@@ -43,11 +45,11 @@ type change[T any] struct {
 
 // Wrapper around a changeSummary that allows comparison
 // of when the change was made.
-type changeSummaryAndIndex struct {
+type changeSummaryAndInsertNumber struct {
```

Comment:
Kind of verbally clunky but I think `InsertNumber` is better than `index` since this isn't actually an index into anything, and I don't want people to think it's an index into `history`. I'm open to other names